projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.spec.ts

```diff
@@ -260,21 +424,33 @@ describe('LinkValueComponent', () => {
       expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
 
       testHostComponent.inputValueComponent.valueFormControl.setValue(res);
-      testHostFixture.detectChanges();
 
       expect(testHostComponent.inputValueComponent.form.valid).toBeTruthy();
+
       expect(testHostComponent.inputValueComponent.valueFormControl.value instanceof ReadResource).toBeTruthy();
+
       const newValue = testHostComponent.inputValueComponent.getNewValue();
       expect(newValue instanceof CreateLinkValue).toBeTruthy();
       expect((newValue as CreateLinkValue).linkedResourceIri).toEqual('http://rdfh.ch/0001/IwMDbs0KQsaxSRUTl2cAIQ');
     });
 
     it('should only create a new value if input is a resource', () => {
+      const valuesSpy = TestBed.get(KnoraApiConnectionToken);
```

Comment:
With all the changes you have applied to the other tests it is hard to see that here is where you have actually fixed the problem. Can you please dedicate one fix PR to exactly the fix instead of also applying multiple changes that are not relevant to the bug/problem at hand?

Reply:
please see https://github.com/dasch-swiss/knora-ui-ng-lib/pull/52#issuecomment-604895476